kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/AbstractRefinableView.java

```diff
@@ -74,6 +75,17 @@ protected StorageKey initialValue() {
     this.entityTest = constraints.toEntityPredicate(accessor);
   }
 
+  protected AbstractRefinableView(AbstractRefinableView<?> view, Schema schema) {
+    this.dataset = (Dataset<E>) view.dataset;
```

Comment:
I don't think this works. If I had a dataset that was instantiated with a specific type, then called asSchema to change the schema, I don't end up with a usable dataset because it still has the same type class. I think this will have to create a copy of the dataset with GenericData.Record as its class.